### ISSUE TITLE
Bug: Trace einsum VJP ("ii->") fails with planning error

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#557]
## Description

Computing the VJP (reverse-mode gradient) of a trace einsum `"ii->"` fails with a planning error. JVP (forward-mode) works correctly for the same operation.

## Reproduction

```rust
use tenferro::{grad, set_default_runtime, GradOptions, RuntimeContext, Tensor};
use tenferro_prims::CpuContext;

let _g = set_default_runtime(RuntimeContext::Cpu(CpuContext::new(1)));
let mut a = Tensor::from_slice(&[1.0_f64, 2.0, 3.0, 4.0], &[2, 2]).unwrap();
a.set_requires_grad(true).unwrap();
let tr = Tensor::einsum("ii->", &[&a]).unwrap();
let grads = grad(&[&tr], &[&a], None, GradOptions::default()).unwrap();
// Error: "failed to plan structured einsum: output label 105 is not present in inputs"
```

## Observed behavior

- Error message: `invalid autodiff argument: invalid AD tensor operands: failed to plan structured einsum: output label 105 is not present in inputs`
- Affects **both real and complex** tensors
- **JVP (forward-mode) works correctly** for the same `"ii->"` subscripts
- Diagonal extraction `"ii->i"` VJP also works correctly

## Likely root cause

The einsum rrule constructs a reverse subscript string where it needs to produce the contracted "i" label in the output. For scalar trace output (`"ii->"`), the reverse subscript generation creates an invalid plan because the output labels of the reverse einsum include labels that were contracted away in the forward pass.

Label `105` is ASCII for `'i'`, confirming the issue is with the reverse subscript for the contracted index.
